.github/workflows/spack_cpu_build.yaml

```diff
@@ -41,8 +38,10 @@ jobs:
           SPACK_SPEC: ${{ matrix.spack_spec }}
         run: |
           ls && pwd
-          . ./tpl/spack/share/spack/setup-env.sh
+          . /opt/spack/share/spack/setup-env.sh
```

Comment:
This is more sustainable, and can be swapped out when we want to use ExaGO's spack. 

It might just make more sense to have a `package.py` stored in the repo for ExaGO as that is the main thing that is regularly modified instead of relying on submodules. Still would be nice to pin a spack version and support that workflow, but making it an optional clone would be best.

Reply:
The spack repo is is incredibly active, I would be hesitant to pin a version of spack because that's just another package we would could get hung up on a "blessed" version or forget to upgrade that one clone...

Reply:
To clarify, for 100% reproducible spack builds we need to pin the version in certain situations (such as profiling runs, or key performance runs).

For CI and automated builds, I do think that just using spack's develop like we do with this base image is the best way to go.

Is that in line with what you were thinking?